spec/models/preneeds/claimant_spec.rb

```diff
@@ -1,33 +1,43 @@
 # frozen_string_literal: true
 require 'rails_helper'
-require 'support/preneeds_helpers'
 
 RSpec.describe Preneeds::Claimant do
-  include Preneeds::Helpers
-
   subject { described_class.new(params) }
 
   let(:params) { attributes_for :claimant }
 
-  it 'populates the model' do
-    expect(json_symbolize(subject)).to eq(xml_dates(params))
-  end
-
   it 'specifies the permitted_params' do
     expect(described_class.permitted_params).to include(
       :date_of_birth, :desired_cemetery, :email, :phone_number, :relationship_to_vet, :ssn
     )
 
     expect(described_class.permitted_params).to include(
-      address: Preneeds::Address.permitted_params, name: Preneeds::Name.permitted_params
+      address: Preneeds::Address.permitted_params, name: Preneeds::FullName.permitted_params
     )
   end
 
-  it 'produces a message hash whose keys are ordered' do
-    expect(subject.message.keys).to eq(
-      [
-        :address, :dateOfBirth, :desiredCemetery, :email, :name, :phoneNumber, :relationshipToVet, :ssn
-      ]
-    )
+  describe 'when converting to eoas' do
+    it 'produces an ordered hash' do
+      expect(subject.as_eoas.keys).to eq(
+        [:address, :dateOfBirth, :desiredCemetery, :email, :name, :phoneNumber, :relationshipToVet, :ssn]
+      )
+    end
+
+    it 'removes :email and :phone_number if blank' do
+      params[:email] = ''
+      params[:phone_number] = ''
+      expect(subject.as_eoas.keys).not_to include(:email, :phoneNumber)
+    end
+
+    it 'converts ssn to eoas format' do
+      expect(subject.as_eoas[:ssn]).to match(/\d{3}-\d{2}-\d{4}/)
```

Comment:
FYI, we are already enforcing this format on the front end, even if the input appears visually to allow a format without dashes. The SSN will be sent with dashes inserted.

Reply:
Good catch @U-DON, its an artifact of the first iteration of Preneeds.